tests/acl/test_acl_outer_vlan.py

```diff
@@ -395,18 +408,50 @@ def check_rule_counters(duthost):
         return True
 
 
-@pytest.fixture(scope="module", autouse=True)
-def teardown(duthosts, rand_one_dut_hostname):
+def check_mac_status(duthost, ptfadapter, vlan, mac, port):
     """
-    Teardown fixture to clean up DUT to initial state
+    Check if mac table has expected vlan and mac
 
     Args:
-        duthosts: All DUTs objects belonging to the testbed
-        rand_one_dut_hostname: Hostname of a random chosen dut to run test
+        duthost: DUT host object
+        ptfadapter: ptfadapter
+        vlan: expected vlan
+        mac: expected mac
+        port: source port
+    Returns:
+        Bool value
     """
-    yield
-    duthost = duthosts[rand_one_dut_hostname]
-    config_reload(duthost, safe_reload=True, check_intf_up_ports=True)
+    # Populate mac table on DUT
+    # Learn mac
+    mac_pkt = testutils.simple_tcp_packet(eth_src=mac, dl_vlan_enable=True, vlan_vid=vlan)
+    testutils.send(ptfadapter, port, mac_pkt)
+    res = duthost.shell("show mac")['stdout_lines']
+    for line in res:
+        data = line.split()
+        if data and len(data) == 5:
+            if data[1] == str(vlan) and data[2].lower().encode() == mac:
```

Comment:
`mac` should be a string. Is the `encode` required?

Reply:
mac is bytes